packages/rollup-full-node/src/app/message-submitter.ts

```diff
@@ -0,0 +1,289 @@
+/* External Imports */
+import { DB } from '@eth-optimism/core-db'
+import {
+  abiDecodeL2ToL1Message,
+  abiEncodeL2ToL1Message,
+  L2ToL1Message,
+} from '@eth-optimism/rollup-core'
+import {
+  add0x,
+  BigNumber,
+  getLogger,
+  logError,
+  ONE,
+  ZERO,
+} from '@eth-optimism/core-utils'
+import { Contract } from 'ethers'
+
+import { L2ToL1MessageSubmitter } from '../types'
+import { TransactionReceipt } from 'ethers/providers/abstract-provider'
+
+const log = getLogger('rollup-message-submitter')
+
+export class NoOpL2ToL1MessageSubmitter implements L2ToL1MessageSubmitter {
+  public async submitMessage(l2ToL1Message: L2ToL1Message): Promise<void> {
+    log.debug(
+      `L2ToL1Message received by NoOpL2ToL1MessageSubmitter: ${JSON.stringify(
+        NoOpL2ToL1MessageSubmitter
+      )}`
+    )
+    return
+  }
+}
+
+/**
+ *  Default Message Submitter implementation. This will be deprecated when message submission works properly.
+ */
+export class DefaultL2ToL1MessageSubmitter implements L2ToL1MessageSubmitter {
+  public static readonly LAST_CONFIRMED_KEY: Buffer = Buffer.from(
+    'last_confirmed'
+  )
+  public static readonly LAST_SUBMITTED_KEY: Buffer = Buffer.from(
+    'last_submitted'
+  )
+  public static readonly LAST_QUEUED_KEY: Buffer = Buffer.from('last_queued')
+
+  private lastSubmitted: BigNumber
```

Comment:
Are we sure we need to use `BigNumber`s here? From a quick calculation we could run a million transactions a second for over 200 years before overflowing a regular JavaScript number: `Number.MAX_SAFE_INTEGER/(60*60*24*365*1000000) => 285.61641472415624` or is there another benefit of BigNumbers that I'm not seeing?